graphql/image_test.go

```diff
@@ -65,6 +65,40 @@ func TestToolchains(t *testing.T) {
 	assert.Equal(t, testToolchain, utility.FromStringPtr(res[0].Name))
 }
 
+func TestEvents(t *testing.T) {
+	config := New("/graphql")
+	ctx := getContext(t)
+	testConfig := testutil.TestConfig()
+	testutil.ConfigureIntegrationTest(t, testConfig, "TestEvents")
+	require.NoError(t, testConfig.RuntimeEnvironments.Set(ctx))
+
+	// Returns the correct number of events according to the limit.
+	imageID := "ubuntu2204"
+	image := model.APIImage{
+		ID: &imageID,
+	}
+	res, err := config.Resolvers.Image().Events(ctx, &image, 5, 0)
+	require.NoError(t, err)
+	assert.Len(t, res, 5)
+
+	// Does not return the same events in different pages.
+	eventAMIs := []string{}
+	for _, event := range res {
+		eventAMIs = append(eventAMIs, utility.FromStringPtr(event.AMIAfter))
+	}
+	res, err = config.Resolvers.Image().Events(ctx, &image, 5, 1)
+	require.NoError(t, err)
+	assert.Len(t, res, 5)
+	for _, event := range res {
+		assert.False(t, utility.StringSliceContains(eventAMIs, utility.FromStringPtr(event.AMIAfter)))
+	}
+
+	// Errors when no image provided.
+	image = model.APIImage{}
```

Comment:
is there a reason we're testing this to have no error? doesn't the above tests coverthat?

Reply:
good point